### PR TITLE
cpx.sh code refactor

### PR DESCRIPTION
**Pull request by ashrude**
I noticed that `cpx.sh` had some opportunity for optimization, so I took it upon myself to refactor it for you.

The new tool is 198.916666667 times faster than previously using a 1gb file, and even more so with larger files. As seen here:

Screenshot from 2023-09-27 19-27-55

As an added benefit, the updated tool is 38% smaller.

### CODE REVIEW

**Comment by hellidox** [comment]
LGTM

**Comment by clemenzi** [comment]
Isn't better creating an alias for the command mv?

The string for `.bashrc` and `.zshrc` is that:
```bash
alias cpx="mv"
```

**Review by hellidox — CHANGES_REQUESTED** [review]
I think `mv $@` would be better

File: utils/cpx.sh [hunk 1]
```diff
@@ -1 +1 @@
-cp $1 $2 && rm -rf $1
+mv $1 $2
```
Comment (blyxyas) [inline utils/cpx.sh:1]: @hellidox that's not a good idea. The best thing to do here is automatically install Rust and do some code generation, like the pros do.

```suggestion
#!/bin/bash
# Check if Rust is installed, if not, install it
if [ ! command -v rustc ]; then
  # Install Rust
  curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | 
sh
fi
  touch move.rs
  echo "fn main() { let _ = std::fs::rename($1, $2); }" > move.rs
  rustc move.rs
  
  # Now, let's move the installed binary to /usr/bin
  sudo mv ./move /usr/bin

# Now, we can just use our newly compiled `move` command
move $1 $2

```

**Review by blyxyas — CHANGES_REQUESTED** [review]
@hellidox that's not a good idea. The best thing to do here is automatically install Rust and write

File: utils/cpx.sh (hunk 1 above)
Comment (hellidox) [reply utils/cpx.sh:1]: Ah, but what if the source and destination are cross-device?

**Pull request closed without merging**